File: src/main/java/br/com/shoppinglist/shopping_list/identity/infrastructure/config/RegistrationVerificationSettingsAdapter.java
package br.com.shoppinglist.shopping_list.identity.infrastructure.config;

import br.com.shoppinglist.shopping_list.identity.application.port.out.RegistrationVerificationSettings;
import org.springframework.stereotype.Component;

@Component
public class RegistrationVerificationSettingsAdapter implements RegistrationVerificationSettings {

	private final VerificationProperties verificationProperties;

	public RegistrationVerificationSettingsAdapter(VerificationProperties verificationProperties) {
		this.verificationProperties = verificationProperties;
	}

	@Override
	public long codeExpirationMinutes() {
		return verificationProperties.codeExpirationMinutes();
	}
}

File: src/identity/infrastructure/config/registration-verification-settings.adapter.ts
import { Injectable } from '@nestjs/common';
import { ConfigService } from '@nestjs/config';
import { RegistrationVerificationSettings } from '../../application/port/out/registration-verification.settings';

@Injectable()
export class RegistrationVerificationSettingsAdapter extends RegistrationVerificationSettings {
  constructor(private readonly config: ConfigService) {
    super();
  }

  codeExpirationMinutes(): number {
    return Number(this.config.get('VERIFICATION_CODE_EXPIRATION_MINUTES', '15'));
  }
}
